### PR TITLE
Model based testing

### PR DESCRIPTION
**Pull request by danbornside**

### CODE REVIEW

**Comment by danbornside** [comment]
same as !2230, but in iohk's repo to enable CI.

**Review by JaredCorduan — APPROVED** [review]
Looks good to me! I think you just need to run ormolu (there is a script in `./scripts/ormolise.sh` that will run it for you if you have nix installed)

File: alonzo/impl/src/Cardano/Ledger/Alonzo/Translation.hs [hunk 1]
```diff
@@ -205,7 +205,7 @@ translateTxOut (Shelley.TxOutCompact addr value) =
 -- (PParamsUpdate era) = (PParams' StrictMaybe era)
 
 translatePParams ::
-  AlonzoGenesis -> Shelley.PParams (MaryEra c) -> PParams (AlonzoEra c)
+  AlonzoGenesis -> Shelley.PParams (MaryEra era) -> PParams (AlonzoEra c)
```
Comment (nc6) [inline alonzo/impl/src/Cardano/Ledger/Alonzo/Translation.hs]: This change doesn't make any sense.

**Review by nc6 — APPROVED** [review]
LGTM. Can you rebase on master?

File: alonzo/impl/src/Cardano/Ledger/Alonzo/Translation.hs (hunk 1 above)
Comment (danbornside) [reply alonzo/impl/src/Cardano/Ledger/Alonzo/Translation.hs]: I guess, `Shelley.PParams` already doesn't make sense to me, it doesn't use it's argument at all; it's not a type family; i don't see why it was as constrained as any of this at all.  Relaxing this constraint allows me to take a fixed `Shelley.PParams` in tests and use it across several eras.

**Comment by JaredCorduan** [comment]
@danbornside the merge will be unblocked once the commit is signed. if you'd rather not do it, I can sign it for you, just let me know.

**Pull request merged**